andrewinsoul/TravelPortal
# setup development environment

**What does this PR do?**
- This PR sets up the development environment

**Description of Task to be completed?**
- Setup virtual environment with pipenv
- Restructure folder structure created by Django command utils
- Created a Django app (TravelPortalSearch)

**How should this be manually tested?**
- Fetch the branch locally by running `git fetch origin setup-environment`
- Activate the virtual environment by running `pipenv shell` (Ensure pipenv is installed on your machine)
- Run `pipenv install` to install all dependencies
- Ensure you have a .env file in your root directory in the format below: 
```
DB_NAME=xxxxxx
DB_USER=xxxxx
DB_PASSWORD=xxxxx
DB_PORT=xxxxx
DB_HOST=xxxxxx
DJANGO_SETTINGS_MODULE='travelportal_api.settings.dev'
SECRET_KEY=xxxxxx
```
- Then run python manage.py migrate. The migrations should be successful
